hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -583,6 +593,11 @@ public abstract AbfsRestOperation append(final String path,
       ContextEncryptionAdapter contextEncryptionAdapter,
       TracingContext tracingContext) throws AzureBlobFileSystemException;
 
+  public abstract AbfsRestOperation appendBlock(final String path,
```

Comment:
I think if before calling these functions we make sure that we are interacting with AbfsBlobCleint, do a typecast to AbfsBlobClient and then call them, we won't need this dummy implementation in DFSClient.

Concern here is that if some how we end up with calling these on DFSClient, that will be a code bug.

Better to add check wherever these are called that they are called with AbfsBlobCleint only, otherwise we can throw an error there itself with relevant message.

Reply:
Same for all three methods that are relevant for BlobCleint only.

Design wise I think also it makes sense that parent has only those methods that are common to both the child, specific implementations should be in children only.

Reply:
taken